corporate/lib/stripe.py

```diff
@@ -1417,9 +1496,27 @@ def process_initial_upgrade(
                 remote_server_legacy_plan.status = CustomerPlan.ENDED
                 remote_server_legacy_plan.save(update_fields=["status"])
 
-            plan = CustomerPlan.objects.create(
-                customer=customer, next_invoice_date=next_invoice_date, **plan_params
-            )
+            if fixed_price_plan is not None:
+                plan = fixed_price_plan
+                plan.next_invoice_date = next_invoice_date
+                plan.charge_automatically = charge_automatically
+                plan.billing_cycle_anchor = billing_cycle_anchor
+                plan.billing_schedule = billing_schedule
+                if not should_schedule_upgrade_for_legacy_remote_server:
+                    plan.status = CustomerPlan.ACTIVE
+                plan.save(
+                    update_fields=[
+                        "next_invoice_date",
+                        "charge_automatically",
+                        "billing_cycle_anchor",
+                        "billing_schedule",
+                        "status",
+                    ]
+                )
```

Comment:
Hmm. Would there be a good way to write this logic using `**plan_params` -- E.g. using `plan.update(**plan_params)`? As written, it's kinda a chore to check that all the parameters we put into `plan_params` are set properly here, and is a maintainability risk that changing `plan_params` above only affects one of the two code paths here.

Reply:
Updated to use `**plan_params` approach.